homeassistant/components/homeassistant/triggers/state.py

```diff
@@ -52,17 +56,24 @@
 )
 
 
-def TRIGGER_SCHEMA(value: Any) -> dict:  # pylint: disable=invalid-name
-    """Validate trigger."""
-    if not isinstance(value, dict):
+async def async_validate_trigger_config(
+    hass: HomeAssistant, config: ConfigType
+) -> ConfigType:
+    """Validate trigger config."""
+    if not isinstance(config, dict):
         raise vol.Invalid("Expected a dictionary")
 
+    registry = er.async_get(hass)
+    config[CONF_ENTITY_ID] = er.async_resolve_entity_ids(
```

Comment:
This changes the original dictionary. We should make a copy.

Reply:
We do change config in the validators, for example replace a template-y string with a template, or wrap a single item in a list?

Reply:
Voluptuous makes copies of anything it validates. Maybe the validators you are looking at are run after a dictionary validation has taken place ?

Reply:
That's the case, fixed now by first passing the data through the schemas, then updating the config[CONF_ENTITY_ID] on the copy.

Reply:
Fixed in 9f1f175